ReactiveCocoa/Swift/Signal.swift

```diff
@@ -410,16 +410,126 @@ public func takeUntil<T, E>(trigger: Signal<(), NoError>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` with history: values of the returned signal
+/// are a tuple whose first member is the previous value and whose second member
+/// is the current value. `initial` is supplied as the first member when `signal`
+/// sends its first value.
+public func combinePrevious<T, E>(initial: T)(signal: Signal<T, E>) -> Signal<(T, T), E> {
+	return signal |> scan((initial, initial)) { previousCombinedValues, newValue in
+		return (previousCombinedValues.1, newValue)
+	}
+}
+
+/// Like `scan`, but sends only the final value and then immediately completes.
+/// If `signal` completes before you observe the resulting signal, the resulting
+/// signal may send no values before completing.
+public func reduce<T, U, E>(initial: U, combine: (U, T) -> U)(signal: Signal<T, E>) -> Signal<U, E> {
+	// We need to handle the special case in which `signal` sends no values.
+	// We'll do that by sending `initial` on the resulting signal (before taking
+	// the last value).
+	let scannedSignal = signal |> scan(initial, combine)
+	let (liftingProducerForScannedSignal, liftingProducerSink: Signal<U, E>.Observer) = SignalProducer.buffer(0)
+	let liftingDisposable = scannedSignal.observe(liftingProducerSink)
```

Comment:
This is a completely valid way to accomplish this, but it seems a little overcomplicated.

Could we use `Signal.pipe()` for the same purpose? That involves similar semantics and gives us similar levels of control, but avoids any need to lift.